src/skmultiflow/meta/dynamic_weighted_majority.py

```diff
@@ -0,0 +1,271 @@
+            List of all existing classes. This is an optional parameter, except
+            for the first partial_fit call, when it becomes obligatory.
+
+        weight: None
+            Instance weight. This is ignored by the ensemble and is only
+            for compliance with the general skmultiflow interface.
+
+        Returns
+        -------
+        DynamicWeightedMajority
+            self
+        """
+        for i in range(len(X)):
+            self.fit_single_sample(
+                X[i:i+1, :], y[i:i+1], classes, weight
+            )
+        return self
+
+    def predict(self, X):
+        """ predict
+
+        The predict function will take an average of the precitions of its
+        learners, weighted by their respective weights, and return the most
+        likely class.
+        
+        Parameters
+        ----------
+        X: Numpy.ndarray of shape (n_samples, n_features)
+            A matrix of the samples we want to predict.
+        
+        Returns
+        -------
+        numpy.ndarray
+            A numpy.ndarray with the label prediction for all the samples in X.
+        
+        """
+        return np.array([np.argmax(self.predict_proba(X))])
+
+    def predict_proba(self, X):
+        """ predict_proba
+        
+        Predicts the probability of each sample belonging to each one of the 
+        known classes.
+        
+        Parameters
+        ----------
+        X: Numpy.ndarray of shape (n_samples, n_features)
+            A matrix of the samples we want to predict.
+        
+        Returns
+        -------
+        numpy.ndarray
+            An array of shape (n_samples, n_features), in which each outer
+            entry is associated with the X entry of the same index.
+        """
+        return self._aggregate_expert_predictions(
+                    self.get_expert_predictions(X))
+
+    def fit_single_sample(self, X, y, classes=None, weight=None):
+        """
+        Fits a single sample of shape X.shape=(1, n_attributes) and y.shape=(1)
+
+        Aggregates all experts' predictions, diminishes weight of experts whose
+        predictions were wrong, and may create or remove experts every _period_
+        samples.
+
+        Finally, trains each individual expert on the provided data.
+
+        Train loop as described by Kolter and Maloof [1]_.
+
+
+        Parameters
+        ----------
+        X: Numpy.ndarray of shape (n_samples, n_features) 
+            Features matrix used for partially updating the model.
+            
+        y: Array-like
+            An array-like of all the class labels for the samples in X.
+            
+        classes: list 
+            List of all existing classes. This is an optional parameter.
+
+        weight: None
+            Instance weight. This is ignored by the ensemble and is only
+            for compliance with the general skmultiflow interface.
+        """
+        self.epochs += 1
+        self.num_classes = max(
```

Comment:
`self.num_classes` is not defined in the `__init__` method

Reply:
it's defined in the ```reset()``` method, which is called in ```__init__```